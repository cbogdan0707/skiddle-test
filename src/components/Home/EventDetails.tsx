import React, { ReactElement } from 'react';
import { Link } from 'react-router-dom';

interface EventProps {
  imageUrl: string;
  eventName: string;
  description: string;
  location: string;
  date: string;
  id: number;
}

export const EventDetails = ({
  eventDetails,
}: {
  eventDetails: EventProps;
}): ReactElement => {
  return (
    <div className="skiddle-c-event-details">
      <img alt="event" src={eventDetails.imageUrl} />
      <div className="skiddle-u-text-align-center skiddle-u-bold skiddle-u-mt-small">
        {eventDetails.eventName}
      </div>
      <div className="skiddle-u-mt-small">{eventDetails.description}</div>
      <div className="skiddle-u-mt-small skiddle-c-location">
        {eventDetails.location}
      </div>
      <div className="skiddle-u-mt-small skiddle-c-date">
        {eventDetails.date}
      </div>
      <div className="skiddle-u-text-align-center skiddle-u-mt-medium">
        <Link
          className="skiddle-u-primary-button skiddle-u-pa-small"
          to={{ pathname: `/event/${eventDetails.id}` }}
        >
          view details
        </Link>
      </div>
    </div>
  );
};
