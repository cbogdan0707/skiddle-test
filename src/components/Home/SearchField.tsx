import React, { ReactElement } from 'react';
import Loader from 'react-loader-spinner';
import { FetchEvents } from '../../types/types';

export const SearchField = ({
  inputText,
  setInputText,
  searchResults,
}: FetchEvents): ReactElement => {
  return (
    <div className="skiddle-u-pb-large skiddle-c-search">
      <input
        data-testid="search-input"
        className="skiddle-c-search-input"
        name="searchEvents"
        value={inputText}
        placeholder="search for events"
        onChange={(e) => setInputText(e.target.value)}
      />
      {searchResults.loading && (
        <span className="skiddle-c-loading-spinner-container">
          <Loader
            type="Oval"
            color="rgb(70, 195, 190)"
            height={24}
            width={24}
          />
        </span>
      )}
    </div>
  );
};
