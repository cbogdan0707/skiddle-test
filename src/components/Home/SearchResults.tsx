import React, { ReactElement } from 'react';
import { EventDetails } from './EventDetails';
import { RawEventsResults } from '../../types/types';

export const SearchResults = ({
  searchResults,
}: {
  searchResults: {
    error: number;
    errormessage: string;
    result: { results: Array<RawEventsResults> };
  };
}): ReactElement => {
  return (
    <>
      {searchResults.error && (
        <div data-testid="fetch-error">Error: {searchResults.errormessage}</div>
      )}
      {searchResults.result &&
        searchResults.result.results &&
        searchResults.result.results.length !== 0 && (
          <section className="skiddle-c-search-results skiddle-u-pt-large">
            <ul className="skiddle-c-search-results-container">
              {searchResults.result.results.map((event: RawEventsResults) => (
                <li key={event.id}>
                  {
                    <EventDetails
                      eventDetails={{
                        id: event.id,
                        imageUrl: event.largeimageurl,
                        eventName: event.eventname,
                        description: event.description,
                        location: event.venue.address,
                        date: event.date,
                      }}
                    />
                  }
                </li>
              ))}
            </ul>
          </section>
        )}
    </>
  );
};
