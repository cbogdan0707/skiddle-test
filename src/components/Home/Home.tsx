import React, { ReactElement } from 'react';
import { SearchResults } from './SearchResults';
import { SearchField } from './SearchField';
import { useDebouncedSearch } from '../../utils/hooks';
import { fetchEvents } from '../../utils/fetchData';
import { EventsResponse } from '../../types/types';

const useSearchEvents = () =>
  useDebouncedSearch(
    (text: string): Promise<EventsResponse> => fetchEvents(text)
  );

export const Home = (): ReactElement => {
  const { inputText, setInputText, searchResults } = useSearchEvents();

  return (
    <div className="App">
      <header className="skiddle-u-background-primary">
        <img
          className="skiddle-c-logo"
          alt="skiddle logo"
          src="https://s3-eu-west-1.amazonaws.com/skiddlecdn-general/assets/logo/png/skiddle-logo-white-landscape.png"
        />
        <SearchField
          inputText={inputText}
          setInputText={setInputText}
          searchResults={searchResults}
        />
      </header>
      <SearchResults searchResults={searchResults} />
    </div>
  );
};
