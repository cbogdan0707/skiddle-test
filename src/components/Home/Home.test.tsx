import React from 'react';
import { render } from '@testing-library/react';
import { Home } from './Home';

describe('Home', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
