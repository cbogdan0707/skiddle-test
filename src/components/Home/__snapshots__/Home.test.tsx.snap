// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="skiddle-u-background-primary"
    >
      <img
        alt="skiddle logo"
        class="skiddle-c-logo"
        src="https://s3-eu-west-1.amazonaws.com/skiddlecdn-general/assets/logo/png/skiddle-logo-white-landscape.png"
      />
      <div
        class="skiddle-u-pb-large skiddle-c-search"
      >
        <input
          class="skiddle-c-search-input"
          data-testid="search-input"
          name="searchEvents"
          placeholder="search for events"
          value=""
        />
        <span
          class="skiddle-c-loading-spinner-container"
        >
          <div
            aria-busy="true"
            class=""
          >
            <svg
              aria-label="audio-loading"
              height="24"
              stroke="rgb(70, 195, 190)"
              viewBox="0 0 38 38"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <g
                  stroke-width="2"
                  transform="translate(1 1)"
                >
                  <circle
                    cx="18"
                    cy="18"
                    r="18"
                    stroke-opacity=".5"
                  />
                  <path
                    d="M36 18c0-9.94-8.06-18-18-18"
                  >
                    <animatetransform
                      attributeName="transform"
                      dur="1s"
                      from="0 18 18"
                      repeatCount="indefinite"
                      to="360 18 18"
                      type="rotate"
                    />
                  </path>
                </g>
              </g>
            </svg>
          </div>
        </span>
      </div>
    </header>
    <section
      class="skiddle-c-search-results skiddle-u-pt-large"
    />
  </div>
</DocumentFragment>
`;
